src/Database/PostgisFunctions/MagellanGeometryProcessingFunctions.php

```diff
@@ -179,6 +179,17 @@ public static function lineMerge($geometry, bool|Expression|\Closure|null $direc
         return MagellanBaseExpression::geometry('ST_LineMerge', [GeoParam::wrap($geometry), $directed]);
     }
 
+    /**
+     * Computes the line which is the section of the input line starting and ending at the given fractional locations. The first argument must be a LINESTRING. The second and third arguments are values in the range [0, 1] representing the start and end locations as fractions of line length. The Z and M values are interpolated for added endpoints if present.
+     *
+     *
+     * @see https://postgis.net/docs/ST_LineSubstring.html
+     */
+    public static function lineSubstring($geometry, float|Expression|\Closure $startFraction, float|Expression|\Closure $endFraction): MagellanGeometryExpression
```

Comment:
Can we quickly change this to match the centroid function in the way this handles geometries and geographies?
Let's also add another optional parameter `geometryType` and pass that to the function call as we do for the centroid.

https://github.com/clickbar/laravel-magellan/blob/8421042561c28bc3730257ef2aa79a878e04f7d4/src/Database/PostgisFunctions/MagellanGeometryProcessingFunctions.php#L79

Reply:
Yes geometry but thats the default with passing null anyway so you can just keep this the same as it is in the centroid function and keep it nullable